example/traffic_mgmt/odp_traffic_mgmt.c

```diff
@@ -293,6 +293,40 @@ static uint8_t  g_print_tm_stats   = TRUE;
 
 static void tester_egress_fcn(odp_packet_t odp_pkt);
 
+static uint64_t tm_shaper_min_rate;
+static uint64_t tm_shaper_max_rate;
+static uint32_t tm_shaper_min_burst;
+static uint32_t tm_shaper_max_burst;
+
+static uint64_t
+clamp_rate(uint64_t rate)
+{
+	uint64_t val = MIN(MAX(rate, tm_shaper_min_rate), tm_shaper_max_rate);
+
+	/* PIR can be zero just with CIR valid and vice versa */
+	if (!rate)
+		return 0;
```

Comment:
Since neither this patch nor any preceding patch has changed, then the problem has has been truly resolved. In general we should not have the correctness of a patch depend on later patches. However, since the issue is related to an existing problem and not introduced by this PR, we can live with it temporarily. But still the "vice versa" part of the comment is not true and AFAICS not needed here and can be removed from the comment. I would also propose adding a remark in the commit message for the benefit of whoever might bump into this commit later:

"Clamp shaper profile rate and burst to limits from capability. Continue the practice of using zero peak rate and zero peak burst to indicate single-rate shaping even though it is not strictly compliant with the API."